src/main/java/com/kadirakinkorkunc/urlshortener/mapper/ConfigurationMapper.java

```diff
@@ -0,0 +1,60 @@
+package com.kadirakinkorkunc.urlshortener.mapper;
+
+import com.kadirakinkorkunc.urlshortener.model.Configuration;
+import com.kadirakinkorkunc.urlshortener.model.ConfigurationRequest;
+import com.kadirakinkorkunc.urlshortener.model.ConfigurationResponse;
+import com.kadirakinkorkunc.urlshortener.util.ConfigurationUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ConfigurationMapper {
+
+    public ConfigurationResponse mapConfigurationToConfigurationResponse(Configuration configuration) {
+        ConfigurationResponse configurationResponse = new ConfigurationResponse();
+        configurationResponse.setKey(configuration.getKey());
+        configurationResponse.setValue(configuration.getValue());
+        return configurationResponse;
+    }
+
+    public List<ConfigurationResponse> mapConfigurationsToConfigurationResponses(List<Configuration> configurationList) {
+        return configurationList.stream().map(this::mapConfigurationToConfigurationResponse).collect(Collectors.toList());
+    }
+
+    public Configuration mapConfigurationReqToConfiguration(ConfigurationRequest configurationRequest) {
+        Configuration configuration = new Configuration();
+        configuration.setKey(configurationRequest.getKey());
+        setConfVal(configuration, configurationRequest.getValue());
+        return configuration;
+    }
+
+    private void setConfVal(Configuration configuration, Object val) {
+        Class acceptableType = configuration.getKey().getType();
+        ConfigurationUtils.checkConfigValueClass(configuration.getKey(), val);
+
+
+        if (String.class == acceptableType) {
+            configuration.setTextVal(val.toString());
+            return;
+        }
+
+        if (Integer.class == acceptableType) {
+            configuration.setIntVal(Integer.valueOf(val.toString()));
+            return;
+        }
+
+        if (Long.class == acceptableType) {
+            configuration.setLongVal(Long.valueOf(val.toString()));
+            return;
+        }
+
+        if (Boolean.class == acceptableType) {
+            configuration.setBoolVal(Boolean.valueOf(val.toString()));
+            return;
+        }
```

Comment:
Instead of these if conditions you can use switch.

```suggestion

        switch(acceptableType) {
            case String.class:
                configuration.setTextVal(val.toString());
                return;
            case Integer.class:
                 configuration.setIntVal(Integer.valueOf(val.toString()));
                 return;
            case Long.class:
                  configuration.setLongVal(Long.valueOf(val.toString()));
                  return;
            case Boolean.class:
                  configuration.setBoolVal(Boolean.valueOf(val.toString()));
                  return;
```

Reply:
"Class" object is not an acceptable argument for a switch statement therefore i used this way to handle the case.